packages/tools/client-debugger/client-debugger-view/src/components/AudienceView.tsx

```diff
@@ -62,6 +62,47 @@ export function AudienceView(props: AudienceViewProps): React.ReactElement {
 		};
 	}, [clientDebugger, audience, setAllAudienceMembers, setAudienceHistory]);
 
+	const audienceClientMetaData: AudienceClientMetaData[] = [];
+
+	for (const [clientId, client] of allAudienceMembers) {
+		audienceClientMetaData.push({ clientId, client });
+	}
+
+	return (
+		<_AudienceView
+			clientId={myClientId}
+			audienceClientMetaData={audienceClientMetaData}
+			onRenderAudienceMember={onRenderAudienceMember}
+			audienceHistory={audienceHistory}
+		/>
+	);
+}
+
+/**
+ * {@link _AudienceView} input props.
+ */
+export interface _AudienceViewProps {
+	clientId: string | undefined;
+	audienceClientMetaData: AudienceClientMetaData[];
+	onRenderAudienceMember: (props: AudienceMemberViewProps) => React.ReactElement;
+	audienceHistory: readonly AudienceChangeLogEntry[];
+}
+
+/**
+ * Audience View displaying current audience members and audience history
+ *
+ * @remarks Operates strictly on raw data, so it can be potentially re-used in contexts that don't have
+ * direct access to the Client Debugger.
+ *
+ * @internal
+ */
+export function _AudienceView(props: _AudienceViewProps): React.ReactElement {
+	const { clientId, audienceClientMetaData, onRenderAudienceMember, audienceHistory } = props;
+
+	const myClientConnection = audienceClientMetaData.find(
```

Comment:
This is only used (I believe) by `MembersView` to compare IDs and determine if the Client is the current user. I think we can update `MembersView` to not require this, and just take in an ID.

Reply:
After our offline chat, I think this change is more complex than I initially thought. Let's hold off on this and do it in a subsequent PR.